dotnet/src/SemanticKernel.Core/KernelBuilder.cs

```diff
@@ -55,110 +90,84 @@ public Kernel Build()
     }
 
     /// <summary>
-    /// Add a logger to the kernel to be built.
+    /// Configures the services to be available through the <see cref="Kernel"/>.
     /// </summary>
-    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
-    /// <returns>Updated kernel builder including the logger.</returns>
-    public KernelBuilder WithLoggerFactory(ILoggerFactory loggerFactory)
+    /// <param name="configureServices">Callback invoked as part of this call. It's passed the service collection to manipulate.</param>
+    /// <returns>This <see cref="KernelBuilder"/> instance.</returns>
+    /// <remarks>The callback will be invoked synchronously as part of the call to <see cref="ConfigureServices"/>.</remarks>
+    public KernelBuilder ConfigureServices(Action<IServiceCollection> configureServices)
```

Comment:
Why `ConfigureServices` instead of just exposing a `Services` collection property?

Reply:
A combination of a) that seemed to be the pattern employed across the various M.E libraries and ASP.NET, and b) it keeps the APIs on KernelBuilder fluent for KernelBuilder.  You think it should change?

Reply:
> that seemed to be the pattern employed across the various M.E libraries and ASP.NET

The past couple releases have gone away with some of that. See https://learn.microsoft.com/en-us/aspnet/core/migration/50-to-60-samples?view=aspnetcore-8.0#add-services.

> b) it keeps the APIs on KernelBuilder fluent for KernelBuilder

This point makes sense, if you want to keep it fluent.

> You think it should change?

I'm not sure. The part that I always found confusing when doing these `ConfigureXXX(action)` methods is "when does my action run? Even here between `ConfigureServices` and `ConfigurePlugins`, the action runs at different times - services runs inline and plugins don't run until `Build` is called.

Following the .NET 6+ ASP.NET / Hosting pattern, this could look more like:

```csharp
class KernelBuilder
{
    IServiceCollection Services { get; }
    PluginBuilder Plugins { get; }
}
```

The usage wouldn't be fluent anymore. And we'd need a special method to handle the `ConfigurePlugins(Action<IServiceProvider, ICollection<IKernelPlugin>> configurePlugins)` case. That one has to be deferred for sure.

Reply:
Thanks, @eerhardt. I'll leave it as ConfigureServices/Plugins for this PR.  One the one hand, I like the simplicity of just .Services/.Plugins. On the other hand, for these use cases I think there's some value in the fluency, in particular when just appending .Build(), we'd still need something special for the ConfigurePlugins (as you say), and there's something to be said for keeping the services stuff a bit more buried.  If we keep them, I'm also wondering if we should name them WithServices and WithPlugins.